org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/internal/workspace/ImportingBrokenProject.groovy

```diff
@@ -50,19 +48,17 @@ class ImportingBrokenProject extends ProjectSynchronizationSpecification {
         }
 
         when:
-        importAndWait(projectDir)
+        tryImportAndWait(projectDir)
 
         then:
-        thrown(BuildException)
         findProject("broken-project_").location.toFile() == projectDir.canonicalFile
     }
 
     def "importing the root project of a broken build fails if the root dir is the workspace root"() {
         when:
-        importAndWait(getWorkspaceDir())
+        tryImportAndWait(getWorkspaceDir())
 
         then:
-        thrown(ImportRootProjectException)
```

Comment:
:x: Shouldn't we check the result's status here and in the tests above?

Reply:
I overlooked it; assertions added.